src/clnt_vc.c

```diff
@@ -312,7 +312,7 @@ clnt_vc_ncreate2(int fd,	/* open file descriptor */
 		if (ct->ct_addr.len)
 			mem_free(ct->ct_addr.buf, ct->ct_addr.len);
 
-		if (xd->refcnt == 1) {
```

Comment:
On Di, 2016-09-06 at 10:24 -0700, Daniel Gryniewicz wrote:
> In src/clnt_vc.c:
> > @@ -312,7 +312,7 @@ clnt_vc_ncreate2(int fd,	/* open file
> descriptor */
> >  		if (ct->ct_addr.len)
> >  			mem_free(ct->ct_addr.buf, ct-
> >ct_addr.len);
> >  
> > -		if (xd->refcnt == 1) {
> I agree that checking for 1 looks like the correct thing to do. If it
> was created, it was initialized to 1 and never modified; if it was
> not created, it was incremented. In either case, it cannot be 0, so
> this change at least introduces dead code, and likely a leak as well.
> A quick scan indicates it should be >1 if it is reused, so the check
> for 1 looks correct.
That's what I thought initially as well that's why I created it like
this,
but rpc_dplx_unref() is decrementing the refcount 
       if (rec) {
                if (rpc_dplx_unref(rec, RPC_DPLX_FLAG_LOCKED))
                        REC_UNLOCK(rec);
        }
for the removal of the reference which was referenced by rec.
That's why we need to check for zero here.
Cheers Swen
> 
> 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
> 
> 
>   > 
>   > 

> 
> 

> 
> 
> 
<span class="-x-evo-quoted"><span class="-x-evo-quote-character">> </span></span>{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/nfs-ganesha/ntirpc","title":"nfs-ganesha/ntirpc","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/nfs-ganesha/ntirpc"}},"updates":{"snippets":[{"icon":"PERSON","message":"@dang in #18: I agree that checking for 1 looks like the correct thing to do.  If it was created, it was initialized to 1 and never modified; if it was not created, it was incremented.  In either case, it cannot be 0, so this change at least introduces dead code, and likely a leak as well.  A quick scan indicates it should be \u003e1 if it is reused, so the check for 1 looks correct."}],"action":{"name":"View Pull Request","url":"https://github.com/nfs-ganesha/ntirpc/pull/18/files/b4254b923aaa4847e6732d4ff893614ac2558388#r77679927"}}}